aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/subscription/RealSubscriptionManager.java

```diff
@@ -205,28 +208,31 @@ public void onError(Exception e) {
                         RealSubscriptionManager.this.removeListener(subObj.subscription, unsubscribeMap.get(subObj));
                         RealSubscriptionManager.this.unsubscribe(subObj.subscription);
                     }
-                    clientConnected.countDown();
+                    allClientsConnectedLatch.countDown();
                 }
             });
-            newClients.add(client);
         }
 
+
         try {
-            clientConnected.await();
+            allClientsConnectedLatch.await();
+            Log.d(TAG, "Made [" + newClients.size() + "] MQTT clients");
 
             synchronized (clients) {
                 // Silence the old clients
-                Log.d(TAG, "Old client count: " + clients);
+                Log.d(TAG, "Muting the old clients [ " + clients.size() + "] in total");
                 for (final SubscriptionClient client : clients) {
                     client.setTransmitting(false);
                 }
 
+                Log.d(TAG, "Unmuting the new clients [" + newClients.size() + "] in total");
                 // Unmute new clients
```

Comment:
should we flip the logic of muting/ unmuting? If we unmute first, worst case would be multiple callback for same message.

Reply:
As discussed, will keep it as is to stay consistent with the iOS SDK.